xla/service/cpu/runtime/custom_call_thunk.cc

```diff
@@ -259,7 +260,8 @@ tsl::AsyncValueRef<Thunk::ExecuteEvent> CustomCallThunk::CallUntypedAPI(
                                   op_buffers_.results_shapes[i].ToString(true),
                                   slice.ToString(), res.opaque());
   }
-  void* out_ptr = results.size() == 1 ? results[0] : results.data();
+
+  void* out_ptr = op_buffers_.is_tuple_result ? results.data() : results[0];
```

Comment:
@ezhulenev I have a question on tuple result. If results[1] (used as scratch buffer) is only used by the custom_call_target, is it safe for concurrent execution of thunks? I am having some segfaults, while I tried using restults[1] as scratch.

Reply:
Yes, every concurrent execution will get its own scratch buffer. I'd recommend to run with asan, it usually shows what's the root cause.

Reply:
Thanks! Is there any chance to have overlap of scratch buffer with any other kind of thunks?

Reply:
Do you want them to overlap and be untouched between thunks? Today scratch buffers can overlap, but if they do overlap, runtime will not run thunks concurrently and will serialize execution (runtime uses buffer conflicts to infer concurrent schedule). If you need to pass “scratch” from one thunk to another, it has to be a “regular” result and argument, and you have to pass it explicitly in HLO

Reply:
Could you please point me to the code, where overlaps scenarios gets serialized execution?